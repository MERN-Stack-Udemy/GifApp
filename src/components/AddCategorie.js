import React, { useState } from "react";
import PropTypes from 'prop-types'


export const AddCategorie = ( {setCategories} ) => {

  const [inputvalue, setInputvalue] = useState(''); // ''

  const handleChange = (e) => {
    setInputvalue(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    if(inputvalue.trim().length > 2){
      setCategories(categories => [...categories, inputvalue])
      setInputvalue( "" );
    }
  }

  return (
    <form onSubmit={handleSubmit}>
      <input 
        type = "text" 
        value = {inputvalue} 
        onChange = {handleChange} />
    </form>
  );

};

AddCategorie.propTypes = {
  setCategories : PropTypes.func.isRequired,
}
