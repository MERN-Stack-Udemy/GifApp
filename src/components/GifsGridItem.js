import React from 'react'

export const GifsGridItem = ( {image, title} ) => {
  
  return (
    <div className="card">
      <img 
        src={image.url}
        alt= {title}
      />
      <p>{title}</p>
    </div>
  )
}
