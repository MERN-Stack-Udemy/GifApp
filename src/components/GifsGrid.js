import React from 'react'
import PropTypes from 'prop-types';

import { useFetchGifs } from '../Hooks/useFetchGifs'
import { GifsGridItem } from './GifsGridItem';

export const GifsGrid = ({ category }) => {

  const {data, loading} = useFetchGifs(category);
  // console.log({data, loading})
  return (
    <>
      <h3  className= "animate__jello">{category}</h3>
      { loading && <p>Loading....</p>}
      <div className="card-grid">
        {data.map( (img) => (
          <GifsGridItem 
            key = {img.id} 
            {...img}
          />
        ))}
        
      </div>
    </>
  )
}

GifsGrid.propTypes = {
  category : PropTypes.string.isRequired,
}