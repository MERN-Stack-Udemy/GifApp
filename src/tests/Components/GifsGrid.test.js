import { shallow } from 'enzyme'
import { GifsGrid } from '../../components/GifsGrid'
import { useFetchGifs } from '../../Hooks/useFetchGifs';
jest.mock('../../Hooks/useFetchGifs')

describe('Test on <GifsGrid {category}/> ', () => {
  
  const category = 'One Punch';
  
  test('should to make match with snapshot component', () => {
    
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });
    
    const wrapper = shallow( <GifsGrid category={category} /> );
    expect( wrapper ).toMatchSnapshot();
  })
  
  test('should displayed items when useFechGifs data loaded', () => {
    
    const gifs = [{
      id: 'ABC',
      url: 'https://localhost:3000/cualquier/imagen.gif',
      title: 'una imagen cualquiera'
    },
    {
      id: 'ABC',
      url: 'https://localhost:3000/cualquier/imagen.gif',
      title: 'una imagen cualquiera'
    },{
      id: 'ABC',
      url: 'https://localhost:3000/cualquier/imagen.gif',
      title: 'una imagen cualquiera'
    }]

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });

    const wrapper = shallow( <GifsGrid category={category} /> );
    expect( wrapper.find('p').exists() ).toBe(false);
    expect( wrapper.find('GifsGridItem').length ).toBe(gifs.length);
  })
  
})
