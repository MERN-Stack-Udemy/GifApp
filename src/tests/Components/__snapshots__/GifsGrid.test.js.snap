// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <GifsGrid {category}/>  should to make match with snapshot component 1`] = `
<Fragment>
  <h3
    className="animate__jello"
  >
    One Punch
  </h3>
  <p>
    Loading....
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
